### PR TITLE
soc: st: Add stm32f030x6

### PR DESCRIPTION
**Pull request by wysman**
Signed-off-by: William MARTIN <william.martin@power-lan.com>

### CODE REVIEW

File: soc/arm/st_stm32/stm32f0/Kconfig.defconfig.stm32f030x6 [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Copyright (c) 2022 William MARTIN <william.martin@power-lan.com>
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_STM32F030X6
+
+config SOC
+	default "stm32f030x6"
+
+config NUM_IRQS
+	default 29
```
Comment (FRASTM) [inline soc/arm/st_stm32/stm32f0/Kconfig.defconfig.stm32f030x6:10]: Could you please also check the soc/arm/st_stm32/stm32f0/Kconfig.defconfig.stm32f030x**4** : 
it is said to be like stm32f030x6 however the last irq entry is 27 for USART1_IRQn in the modules/hal/stm32/stm32cube/stm32f0xx/soc/tm32f030x6.h (assuming NUM_IRQS is 28)

Comment (wysman) [reply soc/arm/st_stm32/stm32f0/Kconfig.defconfig.stm32f030x6:10]: Yes, your right !

It's look like they have another errors in the stm32f030x4 dts (including `zephyr/dts/arm/st/f0/stm32f0.dtsi`)
- stm32f030x4 and stm32f030x6 have only one I2C controller.
- stm32f030x4 and stm32f030x6 do not have the timer 15
- stm32f030x4 and stm32f030x6 do not have the usart2

Do you prefers delete nodes from stm32f030x4 and stm32f030x6 dts, or update stm32f0.dtsi ?
The second solution have a large impact on others DTS file of the F0 family (F3x, F7x F9x).

https://www.st.com/resource/en/datasheet/stm32f030f4.pdf

Comment (erwango) [reply soc/arm/st_stm32/stm32f0/Kconfig.defconfig.stm32f030x6:10]: > Do you prefers delete nodes from stm32f030x4 and stm32f030x6 dts, or update stm32f0.dtsi ?

In this case, deleting nodes would be fine.

File: dts/arm/st/f0/stm32f030X6.dtsi [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+			flash0: flash@8000000 {
+				reg = <0x08000000 DT_SIZE_K(32)>;
+			};
+		};
+	};
+};
+
+/delete-node/ &i2c2;
+/delete-node/ &timers15;
+/delete-node/ &usart2;
```
Comment (erwango) [inline dts/arm/st/f0/stm32f030X6.dtsi:26]: End of line is missing:
```
-:32: WARNING:MISSING_EOF_NEWLINE: adding a line without newline at end of file
#32: FILE: dts/arm/st/f0/stm32f030X6.dtsi:26:
+/delete-node/ &usart2;

- total: 0 errors, 1 warnings, 54 lines checked
```

**Review by erwango — CHANGES_REQUESTED** [review]

**Comment by erwango** [comment]
We don't take fix commit. Please squash both commits together.

**Review by FRASTM — DISMISSED** [review]

File: dts/arm/st/f0/stm32f030X6.dtsi [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+		flash-controller@40022000 {
+			flash0: flash@8000000 {
+				reg = <0x08000000 DT_SIZE_K(32)>;
+			};
+		};
+	};
+};
+
+/delete-node/ &i2c2;
+/delete-node/ &timers15;
```
Comment (FRASTM) [inline dts/arm/st/f0/stm32f030X6.dtsi:25]: can you check: no timers6 & timers7  on stm32f030x6

Comment (wysman) [reply dts/arm/st/f0/stm32f030X6.dtsi:27]: Updated !

**Review by erwango — APPROVED** [review]

**Review by FRASTM — APPROVED** [review]

**Pull request merged**